iothub/device/src/Edge/InstalledCertificateValidator.cs

```diff
@@ -37,10 +38,17 @@ private void SetupCertificateValidation()
 #else
             using var store = new X509Store(StoreName.Root, StoreLocation.CurrentUser);
 #endif
-            foreach (X509Certificate2 cert in _certs)
+            try
+            {
+                foreach (X509Certificate2 cert in _certs)
+                {
+                    store.Open(OpenFlags.ReadWrite);
+                    store.Add(cert);
+                }
+            }
+            finally
             {
-                store.Open(OpenFlags.ReadWrite);
-                store.Add(cert);
+                store.Close();
```

Comment:
We were never calling `Close()` which, especially given on net451 it isn't `IDisposable`, seems like it could be leaking something.

> You should always close an X.509 certificate store after use.

<https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509store.close?view=net-6.0>